java/code/src/com/suse/manager/hub/HubManager.java

```diff
@@ -929,6 +1008,239 @@ public List<IssPeripheral> listRegisteredPeripherals(User user, PageControl pc)
+        // Channels that are not synced
+        List<IssV3ChannelResponse> availableCustomChannels = filterHubChannelsByPeripheral(
+                hubCustomChannels,
+                syncedCustomChannels,
+                IssV3ChannelResponse::getChannelLabel,
+                IssV3ChannelResponse::getChannelLabel,
+                availableFilter
+        );
+        List<IssV3ChannelResponse> availableVendorChannels = filterHubChannelsByPeripheral(
+                hubVendorChannels,
+                syncedVendorChannels,
+                IssV3ChannelResponse::getChannelLabel,
+                IssV3ChannelResponse::getChannelLabel,
+                availableFilter
+        );
+        return new ChannelSyncModel(
+                peripheralOrgs,
+                syncedCustomChannels,
+                syncedVendorChannels,
+                availableCustomChannels,
+                availableVendorChannels
+        );
+    }
+
+    /**
+     * Generic helper function that filters hub/peripheral channels by a label
+     *
+     * @param hubChannels a set of channels from the hub
+     * @param peripheralChannels a set of channels from the peripheral
+     * @param hubLabelExtractor the label get method
+     * @param peripheralLabelExtractor the label get method
+     * @param filterFunction the filter function
+     * @param <H> the hub class
+     * @param <P> the peripheral class
+     * @return the filtered list
+     */
+    private static <H, P> List<H> filterHubChannelsByPeripheral(
+            Set<H> hubChannels,
+            Set<P> peripheralChannels,
+            Function<H, String> hubLabelExtractor,
+            Function<P, String> peripheralLabelExtractor,
+            Function<Set<String>, Predicate<String>> filterFunction
+    ) {
+        Set<String> peripheralLabels = peripheralChannels.stream()
+                .map(peripheralLabelExtractor)
+                .collect(Collectors.toSet());
+        Predicate<String> hubFilter = filterFunction.apply(peripheralLabels);
+        return hubChannels.stream()
+                .filter(hub -> hubFilter.test(hubLabelExtractor.apply(hub)))
+                .toList();
+    }
+
+    /**
+     * Sync the channels from "this" hub to the selected peripheral
+     * @param user the SatAdmin
+     * @param peripheralId the peripheral id
+     * @param channelsId the list of channels id from the hub
+     */
+    public void syncChannelsByIdForPeripheral(User user, Long peripheralId, List<Long> channelsId)
+            throws CertificateException, IOException {
+        ensureSatAdmin(user);
+        IssPeripheral issPeripheral = hubFactory.findPeripheralById(peripheralId);
+        IssAccessToken accessToken = hubFactory.lookupAccessTokenFor(issPeripheral.getFqdn());
+        //TODO: check for parents, ask sync for parents if not already synced
+        List<Channel> channels = ChannelFactory.getSession().byMultipleIds(Channel.class).multiLoad(channelsId);
```

Comment:
I would suggest to create a method in `ChannelFactory` for this and do not invoke `getSession()` directly to perfrom queries, as this is basically bypassing our manager -> factory abstraction.